### PR TITLE
Fix utf-8 char path in windows

### PR DESCRIPTION
**Pull request by npes87184**
The file 'E:\安安\scrcpy-win64-v1.12.1-1-g31bd950\scrcpy-server'
exists, however, it will show msg as follow:

INFO: scrcpy 1.12.1 <https://github.com/Genymobile/scrcpy>
stat: No such file or directory
ERROR: 'E:\安安\scrcpy-win64-v1.12.1-1-g31bd950\scrcpy-server' does not
exist or is not a regular file
Press any key to continue...

This patch fixes it.

Signed-off-by: Yu-Chen Lin <npes87184@gmail.com>

### CODE REVIEW

File: app/src/command.c [hunk 1]
```diff
@@ -208,8 +208,23 @@ process_check_success(process_t proc, const char *name) {
 
 bool
 is_regular_file(const char *path) {
+#ifdef __WINDOWS__
```
[the remark this answers is not in the document]
Comment (npes87184) [reply app/src/command.c]: OK

[the remark this answers is not in the document]
Comment (npes87184) [reply app/src/command.c]: Please see new [version](https://github.com/Genymobile/scrcpy/commit/71db667caf1b0512d08a8728751ef48b1d5216f3).

**Comment by rom1v** [comment]
Thank you.

However, I fail to make it work on Windows.

I declared the environment variable `SCRCPY_SERVER_PATH` (in the system preferences) to `C:\Users\user\Downloads\scrcpy-win64-v1.12.1\论\scrcpy-server` (where I put the file).

When I execute `scrcpy.exe` (built with this patch):
```
C:\Users/user/Downloads/scrcpy-win64-v1.12.1>scrcpy
INFO: scrcpy 1.12.1 <https://github.com/Genymobile/scrcpy>
stat: No such file or directory
ERROR: 'C:\Users\user\Downloads/scrcpy-win64-v1.12.1\?\scrcpy-server' does not exist or is not a regular file
Press any key to continue...
```

(If I rename `论` to some ASCII text, it works)

**Comment by npes87184** [comment]
Thank you for your testing.

There are two problems here.

1. The path sent to `is_regular_file` does not handle non-ascii.
2. The env variable needs to consider non-ascii case.

I have a computer which will get absolute path from `get_server_path`, if the path contains non-ascii text, it will failed to `stat`, this problem fixed by 71db667caf1b0512d08a8728751ef48b1d5216f3.

However, if I set env variable as your testing method, it does not work again. After digging on it, I found the problem is due to the getenv. This problem fixed by 2639c6a5ca2a727a94d59282531f580e195fe9c0.

Thanks.

**Comment by rom1v** [comment]
Oh thank you :+1:

There was a problem with `get_server_path()`, which sometimes returned an allocated string, sometimes not. I fixed it beforehand: 7d5845196e6a7e919d0ec3b5fc53fc7dab8d3222

Then I rebased your commit and edited the last one. Please review [`pr1002`](https://github.com/Genymobile/scrcpy/commits/pr1002) (and test it if you can, I did my own tests, but I'm not sure they are exhaustive).

**Comment by npes87184** [comment]
I have tested on my computer, everything works!
Also, the patches look good to me.

Thank you.

**Comment by rom1v** [comment]
:+1: Thank you very much.

Merged into `dev`.

**Comment by rom1v** [comment]
Released in v1.13.

**Pull request closed without merging**